app/helpers/application_helper.rb

```diff
@@ -16,4 +16,17 @@ def link_to_add_polymorphic_association(*args)
     html_options[:partial] = "#{association}/fields"
     link_to_add_association(name, f, association, html_options)
   end
+
+  def form_navigation_btn(action)
+    text = action == :back ? t('back') : t_title(action)
+    action = :index if action == :back
+    target = { controller: controller_name, action: action }
+    button_type = action == :new ? 'success' : 'default'
+    target[:id] = params[:id] unless action == :index
```

Comment:
These are [Ternary if or Conditional ?:](http://rubylearning.com/satishtalim/simple_constructs.html), not single line conditionals. 
```ruby
def form_navigation_btn(action)
  text = action == :back ? t('back') : t_title(action)
  # equals
  if (action == :back)
    text = t('back')
  else
    text = t_title(action)
  end

  action = :index if action == :back
  # equals
  if(action == :back)
    action = :index
  end

  target = { controller: controller_name, action: action }

  button_type = action == :new ? 'success' : 'default'
  # equals
  if(action == :new )
    button_type = 'success'
  else
    button_type = 'default'
  end

  target[:id] = params[:id] unless action == :index
  single_col_xs button_link(text, target, button_type)
end
```